### ISSUE TITLE
Docker image router 1.51.0 and above returns "an error happened in the worker runtime"

### ISSUE THREAD

**Issue by arsonik** [kamilkisiela/graphql-hive#5557]
While upgrading to 1.51.0 up to latest 
we get this error 
```
docker run -p 1029:1029 \
  --env HIVE_TOKEN="xxx" \
  --env APOLLO_TELEMETRY_DISABLED="true" \
  --env HIVE_CDN_ENDPOINT="xxx" \
  --env HIVE_CDN_KEY="xxx" \
  --mount "type=bind,source=xxx.yaml,target=/dist/config/router.yaml" \
  --rm \
  ghcr.io/kamilkisiela/graphql-hive/apollo-router:apollo_router_update_1.51.0
```

```
Digest: sha256:0dbb1a4383566473095b201c9cf9d3487d93d252698b4832bf28853c544e4eff
Status: Downloaded newer image for ghcr.io/kamilkisiela/graphql-hive/apollo-router:apollo_router_update_1.51.0
INFO: Successfully fetched and saved supergraph from GraphQL Hive
{"timestamp":"2024-08-29T12:57:17.400159422Z","level":"INFO","message":"Apollo Router v1.51.0 // (c) Apollo Graph, Inc. // Licensed as ELv2 (https://go.apollo.dev/elv2)","target":"apollo_router::executable","resource":{}}
{"timestamp":"2024-08-29T12:57:17.400392797Z","level":"INFO","message":"Anonymous usage data collection is disabled.","target":"apollo_router::executable","resource":{}}
{"timestamp":"2024-08-29T12:57:17.400568256Z","level":"INFO","message":"Custom plugins are present. To see log messages from your plugins you must configure `RUST_LOG` or `APOLLO_ROUTER_LOG` environment variables. See the Router logging documentation for more details","target":"apollo_router::executable","resource":{}}
{"timestamp":"2024-08-29T12:57:18.005657798Z","level":"WARN","message":"an error happened in the worker runtime TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string\n\n","target":"router_bridge::worker","resource":{}}
{"timestamp":"2024-08-29T12:57:18.005723089Z","level":"WARN","message":"plan_worker: an unknown error occurred TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string\n\n","target":"router_bridge::worker","resource":{}}
{"timestamp":"2024-08-29T12:57:18.012934048Z","level":"ERROR","message":"panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/router-bridge-0.5.27+v2.8.1/src/worker.rs:104:38:\ncalled `Result::unwrap()` on an `Err` value: TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string\n    at send (worker.js:54:25)\n    at run (worker.js:171:19)","target":"apollo_router::executable","resource":{}}
```

running the same command with   ghcr.io/kamilkisiela/graphql-hive/apollo-router:apollo_router_update_1.50.0 still works

**Comment by kamilkisiela** [comment]
Hi, I'm looking into it right now, thanks

**Comment by kindermax** [comment]
This is probably related 
* https://github.com/kamilkisiela/graphql-hive/issues/4796
* https://github.com/apollographql/router/issues/5084

My error looks like this

```
2024-09-05T13:48:22.770855Z INFO  Running with *development* mode settings which facilitate development experience (e.g., introspection enabled)
2024-09-05T13:48:22.779894Z INFO  Running with *development* mode settings which facilitate development experience (e.g., introspection enabled)
2024-09-05T13:48:22.785638Z INFO  Prometheus endpoint exposed at http://0.0.0.0:9090/metrics
2024-09-05T13:48:23.060837Z WARN  an error happened in the worker runtime TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string


2024-09-05T13:48:23.060889Z WARN  plan_worker: an unknown error occurred TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string


2024-09-05T13:48:23.062152Z ERROR  panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/router-bridge-0.6.0+v2.9.0/src/worker.rs:105:38:
called `Result::unwrap()` on an `Err` value: TypeError: Error parsing args at position 1: serde_v8 error: invalid type; expected: object, got: string
    at send (worker.js:58:25)
    at run (worker.js:186:19)
router-1 exited with code 1
```

**Comment by kamilkisiela** [comment]
Latest image should work, I fixed it in #5566
